core/templates/pages/exploration-editor-page/modal-templates/metadata-version-history-modal.component.ts

```diff
@@ -0,0 +1,235 @@
+      this
+        .versionHistoryService
+        .getForwardMetadataDiffData()
+        .committerUsername
+    );
+  }
+
+  onClickExploreForwardVersionHistory(): void {
+    this.yamlStrs.previousVersionMetadataYaml = '';
+    this.yamlStrs.currentVersionMetadataYaml = '';
+
+    const diffData = this.versionHistoryService.getForwardMetadataDiffData();
+
+    // The explanation for these if-conditions is added in the below function.
+    if (diffData.newMetadata !== null) {
+      this.newMetadata = diffData.newMetadata;
+    }
+    if (diffData.oldMetadata !== null) {
+      this.oldMetadata = diffData.oldMetadata;
+    }
+    this.committerUsername = diffData.committerUsername;
+    this.oldVersion = diffData.oldVersionNumber;
+
+    this.updateLeftPane();
+    this.updateRightPane();
+
+    this.versionHistoryService
+      .decrementCurrentPositionInMetadataVersionHistoryList();
+  }
+
+  onClickExploreBackwardVersionHistory(): void {
+    this.yamlStrs.previousVersionMetadataYaml = '';
+    this.yamlStrs.currentVersionMetadataYaml = '';
+
+    const diffData = this.versionHistoryService.getBackwardMetadataDiffData();
+
+    // Explanation for why diffData.newMetadata can be null:
+    // It is explained in VersionHistoryService as to why the values of
+    // newMetadata or oldMetadata can be null. This is because they are elements
+    // of the list fetchedMetadata whose last element can be null which marks
+    // the end of the version history of the exploration metadata.
+    if (diffData.newMetadata !== null) {
+      this.newMetadata = diffData.newMetadata;
+    }
+    if (diffData.oldMetadata !== null) {
+      this.oldMetadata = diffData.oldMetadata;
+    }
+    this.committerUsername = diffData.committerUsername;
+    this.oldVersion = diffData.oldVersionNumber;
+
+    this.updateLeftPane();
+    this.updateRightPane();
+
+    this.fetchPreviousVersionHistory();
+  }
+
+  fetchPreviousVersionHistory(): void {
+    if (!this.versionHistoryService.shouldFetchNewMetadataVersionHistory()) {
+      this.versionHistoryService
+        .incrementCurrentPositionInMetadataVersionHistoryList();
+      return;
+    }
+    const diffData = this.versionHistoryService.getBackwardMetadataDiffData();
+    if (diffData.oldVersionNumber !== null) {
+      this.versionHistoryBackendApiService.fetchMetadataVersionHistoryAsync(
+        this.contextService.getExplorationId(), diffData.oldVersionNumber
+      ).then((response) => {
+        if (response !== null) {
+          this.versionHistoryService.insertMetadataVersionHistoryData(
+            response.lastEditedVersionNumber,
+            response.metadataInPreviousVersion,
+            response.lastEditedCommitterUsername
+          );
+          this.versionHistoryService
+            .incrementCurrentPositionInMetadataVersionHistoryList();
+        } else {
+          this.alertsService.addWarning(
+            'Could not fetch the version history data due to some reasons. ' +
```

Comment:
Drop "due to some reasons". It does not convey anything.

Reply:
Done.